src/models/heads/segmentation/unet_head.py

```diff
@@ -0,0 +1,58 @@
+"""TorchOK Unet.
+
+Adapted from:
+    https://github.com/qubvel/segmentation_models.pytorch/tree/master/segmentation_models_pytorch/decoders/unet
+Licensed under MIT license [see LICENSE for details]
+"""
+from typing import Tuple, Optional
+
+import torch.nn as nn
+from torch import Tensor
+import torch.nn.functional as F
+
+from src.constructor import HEADS
+from src.models.heads.base import AbstractHead
+
+
+@HEADS.register_class
+class UnetHead(AbstractHead):
+    """Head for Unet architecture."""
+
+    def __init__(self,
+                 in_channels: int,
+                 num_classes: int,
+                 do_interpolate: bool = True,
+                 img_size: Optional[Tuple[int]] = None):
+        """Init UnetHead.
+
+        Args:
+            in_features: Size of each input sample.
+            num_classes: A number of classes for output (output shape - ``(batch, classes, h, w)``).
+            do_interpolate: If ``True`` will interpolate features after forward pass.
+            img_size: Requiered output size of image.
+
+        Raises:
+            ValueError: If "do_interpolate" is True, when img_size is None.
+        """
+        super().__init__(in_channels, num_classes)
+
+        self.in_channels = in_channels
+        self.num_classes = num_classes
+        self.do_interpolate = do_interpolate
+        self.img_size = img_size
+
+        if self.do_interpolate and self.img_size is None:
+            raise ValueError('If "do_interpolate" is True, then img_size must be not None!')
+
+        self.head_features = nn.Conv2d(self.in_channels, self.num_classes, kernel_size=1)
```

Comment:
In previous version of TorchOk it calls `self.last_reduction`. Can we rename it? And also we had parameter 
`self.last_red_before_interpol` with default value = True, you can rename it parameter if you want
```
if self.last_red_before_interpol:
    feats = self.last_reduction(feats)
    feats = F.interpolate(feats, input_image.shape[2:], mode='bilinear', align_corners=True)
            
else:
    feats = F.interpolate(feats, input_image.shape[2:], mode='bilinear', align_corners=True)
    feats = self.last_reduction(feats)
```

Reply:
@PososikTeam `self.last_reduction` is a bad name, it came accidentially